src/erf.jl

```diff
@@ -506,3 +506,28 @@ end
 
 logerfcx(x::Real) = logerfcx(float(x))
 logerfcx(x::AbstractFloat) = throw(MethodError(logerfcx, x))
+
+@doc raw"""
+    logerf(x, y)
+
+Compute the natural logarithm of two-argument error function. This is an accurate version of
+ `log(erf(x, y))`, which works for large `x, y`.
+
+External links: [Wikipedia](https://en.wikipedia.org/wiki/Error_function).
+
+See also: [`erf(x,y)`](@ref SpecialFunctions.erf).
+"""
+function logerf(a::Real, b::Real)
+    if abs(a) ≤ 1/√2 && abs(b) ≤ 1/√2
+        return log(erf(a, b))
+    elseif b > a > 0
+        return logerfc(a) + log1mexp(logerfc(b) - logerfc(a))
+    elseif a < b < 0
+        return logerfc(-b) + log1mexp(logerfc(-a) - logerfc(-b))
+    else
+        return log(erf(a, b))
+    end
+end
+
+two(x::Number) = oftype(x, 2)
+log1mexp(x::Real) = x < -log(two(x)) ? log1p(-exp(x)) : log(-expm1(x))
```

Comment:
```suggestion
log1mexp(x::Real) = x < -log(oftype(x,2)) ? log1p(-exp(x)) : log(-expm1(x))
```

Reply:
Why is that an improvement? If it's a matter of style, I find `two(x)` more readable (similar to `zero(x)`, `one(x)`, etc.).

Reply:
`one` and `zero` are builtins and serve as the identity and zero element for various operations. This is just unnecessary since it's used in exactly one location and doesn't encompass  such meanings.

Reply:
Well, I'm fine either way. So let's remove `two(x)`.